lib/semian/net_http.rb

```diff
@@ -0,0 +1,90 @@
+require 'semian'
+require 'semian/adapter'
+
+module Net
+  ProtocolError.include(::Semian::AdapterError)
+
+  class SemianError < ::Net::ProtocolError
+    def initialize(semian_identifier, *args)
+      super(*args)
+      @semian_identifier = semian_identifier
+    end
+  end
+
+  ResourceBusyError = Class.new(SemianError)
+  CircuitOpenError = Class.new(SemianError)
+end
+
+module Semian
+  module NetHTTP
+    include Semian::Adapter
+
+    ResourceBusyError = ::Net::ResourceBusyError
+    CircuitOpenError = ::Net::CircuitOpenError
+
+    DEFAULT_SEMIAN_OPTIONS = {
```

Comment:
Whereas the other adapters you can do a `Mysql2::Client.new(..., semian_options: { ... } )` you can't do it here since you may not be actually making the `Net::Http` instance